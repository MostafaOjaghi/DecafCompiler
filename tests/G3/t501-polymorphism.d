class Animal {
    void makeSound() {

    }
}

class Dog extends Animal {
    void makeSound() {
        Print("bark!");
    }
}

class Cat extends Animal {
    void makeSound() {
        Print("miaw!");
    }
}

void main() {
    Animal[] animals;
    int i;
    int n;

    n = ReadInteger();
    animals = NewArray(n, Animal);
    for (i = 0; i < n; i = i + 1) {
        if (i % 2 == 0) {
            animals[i] = new Cat;
        } else {
            animals[i] = new Dog;
        }
    }

    for (i = 0; i < n; i = i + 1) {
        animals[i].makeSound();
    }
}