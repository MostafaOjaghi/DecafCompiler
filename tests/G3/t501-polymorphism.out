miaw!
bark!
miaw!
bark!
miaw!
bark!
miaw!
bark!
miaw!
bark!